CWE.Data/DataAccessLayer.cs

```diff
@@ -132,5 +134,122 @@ public async Task<IEnumerable<Request>> GetRequests()
+                .Where(x => x.Name == tagName)
+                .SingleOrDefaultAsync();
+
+            if (tag == null)
+            {
+                throw new ArgumentException("The name provided does not match an existing tag in the database.");
+            }
+
+            return tag;
+        }
+
+        /// <summary>
+        /// Creates a tag.
+        /// </summary>
+        /// <param name="name">The name that the tag should have.</param>
+        /// <param name="ownerId">The owner's user ID of the tag.</param>
+        /// <param name="content">The content that the tag holds.</param>
+        /// <returns><see cref="void"/>.</returns>
+        public async Task CreateTagAsync(string name, ulong ownerId, string content)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == name)
+                .AnyAsync();
+            if (tag)
+            {
+                throw new Exception("A tag with this name already exists.");
+            }
+
+            this.dbContext.Tags.Add(new Tag
+            {
+                Name = name,
+                OwnerId = ownerId,
+                Content = content,
+            });
+            await this.dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Deletes a tag from the database.
+        /// </summary>
+        /// <param name="tagName">The name of the tag to delete.</param>
+        /// <returns><see cref="void"/>.</returns>
+        public async Task DeleteTagAsync(string tagName)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == tagName)
+                .SingleOrDefaultAsync();
+
+            if (tag == null)
+            {
+                throw new ArgumentException("The tag provided was not found in the database.");
+            }
+
+            this.dbContext.Tags.Remove(tag);
+            await this.dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Edits the content of a tag, and modifies it to the new content.
+        /// </summary>
+        /// <param name="tagName">The tag to edit.</param>
+        /// <param name="executorId">The user who is attempting to modify the tag.</param>
+        /// <param name="newContent">The content that the new tag should hold.</param>
+        /// <returns><see cref="void"/>.</returns>
+        public async Task EditTagContentAsync(string tagName, ulong executorId, string newContent)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == tagName)
+                .SingleOrDefaultAsync();
+            if (tag == null)
+            {
+                throw new ArgumentException("The tag provided does not match any currently in the database.");
+            }
+
+            if (tag.OwnerId != executorId)
+            {
+                throw new InvalidOperationException("You cannot modify a tag that you do not own.");
+            }
```

Comment:
```suggestion
        /// <param name="newContent">The content that the new tag should hold.</param>
        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
        public async Task EditTagContent(string tagName, string newContent)
        {
            var tag = await this.dbContext.Tags
                .FirstOrDefaultAsync(x => x.Name == tagName);
                
            if (tag == null)
            {
                throw new ArgumentException("The tag provided does not match any currently in the database.");
            }
```
Proper return summary, different method name and the owner check should be done prior to running this. This is the data access layer, it is only responsible for managing data, not checking if all conditions (whether or not the edit is allowed) are being met.